aten/src/ATen/native/cuda/IndexKernel.cu

```diff
@@ -121,7 +121,32 @@ Tensor & masked_select_out_cuda(Tensor & result, const Tensor & self, const Tens
   return masked_select_out_cuda_impl(result, self, mask);
 }
 
+template <typename scalar_t, typename mask_t>
+void gpu_masked_fill_kernel(TensorIterator& iter, scalar_t value) {
+  constexpr bool is_mask_bool = std::is_same<mask_t, bool>::value;
+  gpu_kernel(iter, [=] GPU_LAMBDA (scalar_t input, mask_t mask_value) -> scalar_t {
+      if (!is_mask_bool) {
+        CUDA_KERNEL_ASSERT(mask_value == 0 || mask_value == 1);
+      }
+      return mask_value ? value : input;
+  });
+}
+
+void masked_fill_kernel(TensorIterator& iter, Scalar value) {
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(at::ScalarType::Bool, at::ScalarType::Half, at::ScalarType::BFloat16,
+    iter.dtype(), "masked_fill_cuda", [&] {
+      scalar_t scalar_val = value.to<scalar_t>();
+      auto mask_dtype = iter.input_dtype(1);
+      if (mask_dtype == at::ScalarType::Bool) {
```

Comment:
Please keep ```TORCH_WARN("masked_fill_ received a mask with dtype torch.uint8, this behavior is now deprecated," \	
            "please use a mask with dtype torch.bool instead.");```

Maybe change it to TORCH_WARN_ONCE

Reply:
Or better move this warning to common (CPU,CUDA) code.

Reply:
`masked_fill_impl` in `ATen/native/TensorAdvancedIndexing.cpp` warns.
https://github.com/pytorch/pytorch/pull/38061/files#diff-5312d553f68ae8e12304275aab5ac43cR574-R575